apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -595,6 +619,57 @@ impl Telemetry {
                 .unwrap_or_default()
                 .to_string(),
         );
+        if let Some(metrics_conf) = &config.metrics {
+            // List of custom attributes for metrics
+            let mut attributes: HashMap<String, String> = HashMap::new();
+            if let Some(operation_name) = &req.originating_request.body().operation_name {
+                attributes.insert("operation_name".to_string(), operation_name.clone());
+            }
+
+            if let Some(MetricsCommon {
+                additionnal_metric_labels:
+                    Some(MetricsLabelsConf {
+                        propagate_headers: Some(propagate_headers),
+                        ..
+                    }),
+                ..
+            }) = &metrics_conf.common
+            {
+                for propagate_header in propagate_headers {
```

Comment:
are parts of this factorizable ?

Reply:
if you're speaking about the header_propagation plugin, then I'm not sure, it's not the same structure, not always the same logics.